### PR TITLE
[boost-modular-build-helper] Add BSD and FreeBSD target-os

### PR DESCRIPTION
**Pull request by noeffn**
- #### What does your PR fix?
  Fixes b2 using the wrong operating system on bsd/freebsd resulting in cstddef not being found 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by noeffn** [comment]
Okay this will probably not work like this https://github.com/Kitware/CMake/blob/master/Modules/Platform/FreeBSD.cmake
I looked at this and thought it would be included on BSD

**Comment by noeffn** [comment]
I now use CMAKE_SYSTEM_NAME altough I dont know if it would be better to use VCPKG_CMAKE_SYSTEM_NAME set in the triplet

File: ports/boost-modular-build-helper/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "boost-modular-build-helper",
   "version": "1.80.0",
+  "port-version": 2,
```
Comment (JonLiu1993) [inline ports/boost-modular-build-helper/vcpkg.json:4]:
```suggestion
  "port-version": 1,
```

File: versions/b-/boost-modular-build-helper.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "7ce7f8dca95eb5e2111f586ba7a9a3df46d67c4b",
+      "version": "1.80.0",
+      "port-version": 2
+    },
+    {
+      "git-tree": "20a1203d52eecd52b803063638e91429883302e5",
+      "version": "1.80.0",
+      "port-version": 1
+    },
```
Comment (JonLiu1993) [inline versions/b-/boost-modular-build-helper.json:12]:
```suggestion
```

Comment (JonLiu1993) [reply versions/b-/boost-modular-build-helper.json:12]: @noeffn ,Thanks for your pr, please remove this code and change the `port-version` to 1, then run the command "./vcpkg x-add-version boost-modular-build-helper --overwrite-version" and commit again

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by noeffn** [comment]
Done

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/boost-modular-build-helper/CMakeLists.txt [hunk 3]
```diff
@@ -79,6 +79,8 @@ elseif(APPLE)
     endif()
 elseif(ANDROID)
     list(APPEND B2_OPTIONS target-os=android)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
```
Comment (JavierMatosD) [inline ports/boost-modular-build-helper/CMakeLists.txt:82]:
```suggestion
elseif(VCPKG_DETECTED_CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")

```

**Review by JavierMatosD — CHANGES_REQUESTED** [review]

File: versions/b-/boost-modular-build-helper.json [hunk 4]
```diff
@@ -1,7 +1,8 @@
 {
   "versions": [
     {
-      "git-tree": "537dbfdc2ebfe2ac2c445503c0930b4e00976864",
+
```
Comment (noeffn) [inline versions/b-/boost-modular-build-helper.json:4]: bad merge

File: ports/boost-modular-build-helper/CMakeLists.txt (hunk 3 above)
Comment (noeffn) [reply ports/boost-modular-build-helper/CMakeLists.txt]: Done

**Review by JonLiu1993 — APPROVED** [review]

**Review by JavierMatosD — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thanks!

**Pull request merged**